### DISCUSSION TITLE
If condition causes undefined

### DISCUSSION THREAD

**Discussion by landoDanziger** [maizzle/framework#842] [Q&A]
I have a strange problem - tI'll try and provide a repo in a few weeks - but if i use the same variable name twice - the second one becomes undefined.

If i remove the conditional tags on the second one - it renders the value from my env,config

```
<if condition="page.env === 'esp1'">
<img class="w-full" src="{{ page.bannerContentImageUrl || 'https://via.placeholder.com/624x300' }}" alt="{{ page.bannerContentImageDescription || 'Banner image description' }}" width="624" />
</if>

<if condition="page.env === 'esp2'">
<img class="sm:w-full xs:hidden" src="{{ page.bannerContentImageUrl || 'https://via.placeholder.com/624x300' }}" alt="{{ page.bannerContentImageDescription || 'Banner image description' }}" width="624" />
</if>
```

I tried duplicating the variable name - but still ends up undefined. I think i see related issues in the closed tickets. This items are in side a component being referenced.

**Comment by cossssmin** [comment]
Conditions are basic JavaScript expressions in Maizzle, you are setting `page.env` to that value when using a single `=`. 

To check for equality, use double or triple `===`:

```
<if condition="page.env === 'esp1'">
```

**Reply by landoDanziger** [reply]
sorry - i should have copied it direct - I am tired 

        <if condition="page.env === 'braze'">
    <tr>
      <td class="px-s pt-xs pb-m s" align="center" bgcolor="{{{ page.bannerContentVoucherBackground || '#FB2815' }}}">
        {{ page.bannerContentText || ' words.' }}
      </td>
    </tr>
  </if>``
  
  When used in a component the IF seems to cause the values not to be rendered in the second instance - but renders it in the first one.
  
  I am trying to maintain one template deployed in to 3 different ESPs - for one reason or another it is hard work.
  
  My QA process involves searching for the word 'undefined' in my output :D  - I would prefer for someone to help me turn it into a standalone application and just build the emails completely outside the esp - but.....thats not happening soon

**Reply by landoDanziger** [reply]
Answer seems to be if using lots of `<IF>` that it is good to reset it through `<ELSEF>` and `<ELSE>`
 As there is no `<ENDIF>` closer. Seems it confuses the compiler.

So no more  writing endless: 

```
<IF condition="page.env === 'esp1'>something</IF>
<IF condition="page.env === 'esp2'>something</IF>
<IF condition="page.env === 'esp3'>something</IF>
<IF condition="page.env === 'esp4'>something</IF>
```

I was using regex to strip ESP syntax from different builds, but thought it might be nice for other people to have them labelled with the production environment.

```
<IF condition="page.env === 'esp1'>something</IF>
<ELSEIF condition="page.env === 'esp2'>something</IFELSE>
<ELSE></ELSE>

```
before starting it again 

```
<IF condition="page.env === 'esp1'>more</IF>
<ELSEIF condition="page.env === 'esp2'>more</IFELSE>
<ELSE></ELSE>
```

This has been haunting me for about 6 months....:D I am a such a fool. Ampscript is completley fine with IF ENDIF IF ENDIF - because it has the ENDIF - i got lax and forgot if cant just use `<IF></IF>` and treat it as self closing.

**Reply by landoDanziger** [reply]
For Braze liquid {{ }} ( i may substitute and replace these handlebars entirely and swap them back later ) but - depending on how deep the element is nested in the components - seems to require more @@ ( as far as i can tell )

**Reply by cossssmin** [reply]
I'm currently looking at refactoring the Components system, the new one may solve many of these issues.

By the way, in your situation maybe a [`<switch>`](https://maizzle.com/docs/tags#switch) is better suited?

**Reply by landoDanziger** [reply]
I have only recently realised I should be using switch by now. Slow realisation.

**Reply by landoDanziger** [reply]
I managed to create a different problem - in the braze build - if i switch on uglify - most of my variables become undefined >
edit: thank the lord - the framework update fixed it. Phew

**Reply by landoDanziger** [reply]
I just keep my idiocy here for other people - just incase anyone is as misguided as me - I don't know how to mark as answered.
